software/contrib/egressus_melodiam.md

```diff
@@ -0,0 +1,101 @@
+# Egressus Melodiam (Stepped Melody)
+
+author: Nik Ansell (github.com/gamecat69)
+date: 02-Feb-24
+labels: clocked lfo, sequencer, CV, randomness
+
+Clockable and free-running LFO and random CV pattern generator.
+Six different wave shapes either define the LFO shape or the slew shape between CV pattern steps.
+Use it to generate variable length melodies (using a quantizer), LFOs or wierd and wonderful control voltages.   
+
+# Inputs, Outputs and Controls
+
+digital_in: clock in
+analog_in: not used
+
+knob_1: Set global CV pattern length / free-running clock speed (A pattern length of 1 produces an cycling LFO)
+knob_2: Set LFO / CV Pattern clock division per output
+
+button_1:
+- Short Press (< 300ms): Cycle through slew/LFO shapes for the selected output
+- Long Press (Between 2 and 5 seconds): Generate a new random CV pattern for the selected output
+
+button_2:
+- Short Press (< 300ms): Cycle through outputs to edit
+- Long Press (Between 2 and 5 seconds): Toggle between clocked mode and free-running mode
+
+output_1: LFO / randomly generated CV pattern (0 to +5V)
+output_2: LFO / randomly generated CV pattern (0 to +5V)
+output_3: LFO / randomly generated CV pattern (0 to +5V)
+output_4: LFO / randomly generated CV pattern (0 to +5V)
+output_5: LFO / randomly generated CV pattern (0 to +5V)
+output_6: LFO / randomly generated CV pattern (0 to +5V)
+
+# Getting started
+
+1. Patch a 50% duty cycle square wave or clock with pulses >= 18ms in duration  into the digital input
+2. Connect one or more outputs to an input on another module (e.g. CV modulation inputs)
+3. Select a pattern pength using Knob 1
+4. Start your clock. Each output will now send a random looping CV to your module!
+
+So, what happened in the above example?
+When the module first powered on it automatically generated 6 x 32 step random CV patterns - one for each of the 6 outputs.
+Each time a clock is received, the step advances by one step and then loops when it get to the end of the pattern.
+The length of the pattern loop is controlled using knob 1, which supports a value from 1 to 32.
+
+# Changing the wave shape and clock division of an output

```

Comment:
[required]
Could you please put a copy of your wave shape images (as PNG files) in `software/contrib/egressus_melodiam_docs/*.png` and then add the images into your readme? For people not well-versed in math it may be hard to visualize the waveforms, so having the names & pictures in the documentation would be helpful.

Please see https://github.com/Allen-Synthesis/EuroPi/blob/27bdb9ed84f3879f6ac356ebd62db81716d31b96/software/contrib/pams.md?plain=1#L130-L141 for an example of how to include images in markdown files.

Reply:
done